File: code/c/ds/stack.cpp
#include<iostream>
using namespace std;

int a[100],top=-1;

void pop() {
  if (top==-1)
    cout<<"Stack empty so cant pop man";
  else{
    top--;
    cout<<"its done man";
  }
}

void push() {
  int elem;
  cout<<"enter element to be pushed in stack"<<endl;
  cin>>elem;
  top++;
  a[top] = elem;


}

void print() {
  for(int i=0 ;i<=top; i++) {
    cout<<a[i]<<"\n";
  }
}

int main() {
  int n, elem;
  int out=0;
while(1){
  cout<<"enter the operation you wanna perform"<<endl;
  cout<<"1.Push    2.Pop    3.Print  4. Exit"<<endl;
  cin>>n;


    if (n==1)
      push();
    if (n==2)
      pop();
    if (n==3)
      print();
    if (n==4)
      break;
  }


}

File: code/c/ds/stack.go
package main
import(
    "fmt"

)

//var top int
//top = -1
var a []int

func pop()  {
  if len(a)==0 {
    fmt.Println("no elem in stack to pop")
  }else{
    a = a[0:len(a)-1]
  }
}

func push()  {
  var x int
  fmt.Println("enter elem to be pushed")
  fmt.Scan(&x)
  a = append(a,x)
}

func print()  {

    fmt.Println(a)

}

func main() {
  var choice int
  p:=1
  for p!=0{
    fmt.Println("enter choice")
    fmt.Println("1.Push   2.Pop   3.Print   4.Exit")
    fmt.Scan(&choice)

    if choice==1{
      push()
    }else if choice==2{
        pop()
      }else if choice==3 {
        print()
      }else if choice==4{
        p = 0
      }
    }


}
